plugin/src/main/java/com/iridium/iridiumcore/utils/ItemStackUtils.java

```diff
@@ -81,8 +79,18 @@ public static ItemStack makeItem(Item item) {
             if (item.material == XMaterial.PLAYER_HEAD && item.headData != null) {
                 return setHeadData(item.headData, itemstack);
             } else if (item.material == XMaterial.PLAYER_HEAD && item.headOwner != null) {
+                UUID uuid;
```

Comment:
I dont really like the look of this code, would something like

```java
OfflinePlayer offlinePlayer = Bukkit.getServer().getOfflinePlayer(item.headOwner);
if(!offlinePlayer.hasPlayedBefore()){
	UUID uuid = SkinUtils.getUUID(item.headOwner);
	if(uuid == null){
		return setHeadData("eyJ0ZXh0dXJlcyI6eyJTS0lOIjp7InVybCI6Imh0dHA6Ly90ZXh0dXJlcy5taW5lY3JhZnQubmV0L3RleHR1cmUvZWI3YWY5ZTQ0MTEyMTdjN2RlOWM2MGFjYmQzYzNmZDY1MTk3ODMzMzJhMWIzYmM1NmZiZmNlOTA3MjFlZjM1In19fQ==", itemstack);
	}
	return setHeadData(SkinUtils.getHeadData(uuid), itemstack);
}else{
        return setHeadData(SkinUtils.getHeadData(offlinePlayer.getUniqueId()), itemstack);
}
```
look better?

Reply:
Whatever you want, it's your plugin :)